### PR TITLE
fix(drm): eliminate use of non-existent lv_api_map.h and enable smoke tests

### PR DESCRIPTION
**Pull request by Johennes**
### Description of the feature or fix

Compiling the DRM display driver currently fails because it uses a non-existent include.

```
FAILED: unl0kr.p/.._lvgl_src_drivers_display_drm_lv_linux_drm.c.o 
cc -Iunl0kr.p -I. -I.. -I../.. -I/usr/include/libdrm -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wpedantic -O0 -g '-DUL_VERSION="2.0.2"' -DLV_USE_LINUX_DRM=1 -MD -MQ unl0kr.p/.._lvgl_src_drivers_display_drm_lv_linux_drm.c.o -MF unl0kr.p/.._lvgl_src_drivers_display_drm_lv_linux_drm.c.o.d -o unl0kr.p/.._lvgl_src_drivers_display_drm_lv_linux_drm.c.o -c ../../lvgl/src/drivers/display/drm/lv_linux_drm.c
../../lvgl/src/drivers/display/drm/lv_linux_drm.c:12:10: fatal error: ../../../lv_api_map.h: No such file or directory
   12 | #include "../../../lv_api_map.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~
```

This PR simply removes the include.

I also tried enabling compilation of both the DRM and FBDEV drivers on the CI in this PR.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.10](https://github.com/szepeviktor/astyle/releases/tag/v3.4.10) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: tests/src/lv_test_conf_full.h [hunk 1]
```diff
@@ -101,6 +101,9 @@
 
 #define LV_CACHE_DEF_SIZE       (10 * 1024 * 1024)
 
+#define LV_USE_LINUX_DRM    1
```
Comment (XuNeo) [inline tests/src/lv_test_conf_full.h]: Is it possible to make the test can run on mac?
I  have made a try in https://github.com/lvgl/lvgl/pull/5672. Hope it's the way.

Comment (Johennes) [reply tests/src/lv_test_conf_full.h]: Good call. I tried to mimic your approach here.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -261,6 +261,15 @@ include_directories(${LIBINPUT_INCLUDE_DIRS})
 find_package(PkgConfig)
 pkg_check_modules(xkbcommon REQUIRED xkbcommon)
 
+# libdrm is required for the DRM display driver test case
+include(${CMAKE_CURRENT_LIST_DIR}/FindLibDRM.cmake)
+if(Libdrm_FOUND)
+include_directories(${Libdrm_INCLUDE_DIRS})
```
Comment (XuNeo) [inline tests/CMakeLists.txt]: Please add indent like elsewhere.

**Comment by XuNeo** [comment]
Hi,
I resolved the conflict and check on mac. Now it's working. My changes are [here](https://github.com/lvgl/lvgl/compare/master...XuNeo:lvgl:feature/eliminate-use-of-lv-api-map?expand=1). I can push to this PR if you allow me.

**Comment by Johennes** [comment]
> Hi, I resolved the conflict and check on mac. Now it's working. My changes are [here](https://github.com/lvgl/lvgl/compare/master...XuNeo:lvgl:feature/eliminate-use-of-lv-api-map?expand=1). I can push to this PR if you allow me.

Thanks. I think I messed up because I couldn't get your branch to merge cleanly onto mine. But I've resolved the conflicts manually which hopefully should yield the same result.

**Comment by XuNeo** [comment]
I added guard to macros overrides in cmake. Confirmed it compiles. :)

**Comment by Johennes** [comment]
Great, thank you! :+1:

**Review by XuNeo — APPROVED** [review]

**Review by kisvegabor — APPROVED** [review]
It's great that DRM is tested now at least by a build test.

**Pull request merged**